### ISSUE TITLE
Project roadmap and help with the rust backend

### ISSUE THREAD

**Issue by gokulsoumya** [itsfarseen/breathe-india#2]
First of all, thank you for taking the initiative and making an app which could be potentially useful for a lot of people. I would like to help in whatever way I can with the rust backend. Is there any feature in the works that I can contribute to ? Also what is the project roadmap as of now ?

**Comment by itsfarseen** [comment]
Hi @gokulsoumya :wave: 
Thanks for your kind words and appreciate your interest in helping with the further development of the app. 
As of now, what we badly need is awareness among people, especially people who can be suppliers of goods, to upload the information of what they have to the app.  
Once more people start using the app, we may discover rough edges those need to be smoothed or features those can greatly help their workflow. Other than that as of now I don't have any plans for the backend or frontend.  

But, here are some useful things that you could do:

`google_jwt` module contains code that I took from a medium post to verify the validity of the JWT returned by Google OAuth. It would be useful if you can clean that up and package into a crate, so that the next person who is developing a similar thing can use it.

Currently, the app doesn't have any automated tests, neither unit nor integration. I'm relying on SQLx's compile time verification to ensure the SQL queries won't blow up during runtime, and rest of the functionality is manually tested. This is feasible because the app is relatively small now. It would be useful if you could try to add tests for the backend (also frontend if possible). And it would be awesome if you could describe your experience in a blog post. I'm not sure how easy it would be considering you have to interact with Google to get a sign in token. May be we could make a stub server that hands out a crafted token and use that during testing. Once done, we can set that up in a CI server so that tests are run on every PR. 

Turn the frontend into a PWA. For starters we need to make the browsers show the add to home screen prompt. Another nice thing to have is to cache last few queries and show them even when there is no internet. 

Overall, other than minor things that help the UX, I would prefer not to add major features to the app in the interest of keeping it simple; both to use and to understand the code.

**Comment by itsfarseen** [comment]
What we need the most at the moment is skilled people capable of organizing a drive so that people who can supply things add the information to the app, and show them how easy it is to update the information in the app.

**Comment by itsfarseen** [comment]
See this too https://github.com/itsfarseen/breathe-india/issues/3#issue-875004255

**Comment by gokulsoumya** [comment]
I'll try to add some tests for the backend and also work on refactoring it. As for the server issue I think we can mock the calls for Google auth and use dummy tokens.

Regarding adoption of the app by the people, absence of actual suppliers at least initially would be a barrier. I can spread the news in my college and other circles and I'm sure more and more people will come to know of it. But IMO we need at least two or three actual posts in the needs and supplies section. Seeing placeholder values might make people think that the app isn't usable yet.

**Comment by itsfarseen** [comment]
Yes we are working on it. It would be super helpful if you can share in your college and get people to sign up.
